src/Util/StreamInfo.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace App\Util;
+
+/**
+ * A utility class to get various information from a file stream.
+ */
+class StreamInfo
+{
+    /**
+     * Get the hash for file stream.
+     *
+     * @param array $fileStream The file stream array.
+     *
+     * @return string The hash calculated from the stream.
+     */
+    public static function calculateHash(array $inputFileStream, string $algo = 'sha256') :string
```

Comment:
Can use constant DatasetSubmission::SHA256

Reply:
Does it exist? Is it better to have the constant in this class?